src/disk_io_thread.cpp

```diff
@@ -1087,10 +1104,6 @@ namespace libtorrent
 		TORRENT_ASSERT(j->next == 0);
 		TORRENT_ASSERT((j->flags & disk_io_job::in_progress) || !j->storage);
 
-		mutex::scoped_lock l(m_cache_mutex);
-
-		check_cache_level(l, completed_jobs);
-
 		DLOG("perform_job job: %s ( %s%s) piece: %d offset: %d outstanding: %d\n"
```

Comment:
ssiloti pointed out that you cannot get torn 32 bit reads on the x86 architecture, because it provides a stronger cache coherency guarantees than C++ does. I would be surprised if you can get torn 64 bit reads with modern CPUs as well, but that's not the case here anyway, because this variable is 32 bits.